radio/src/gui/480x272/lcd.cpp

```diff
@@ -557,3 +557,272 @@ void DMABitmapConvert(uint16_t * dest, const uint8_t * src, uint16_t w, uint16_t
   }
 }
 #endif
+
+//OW
+//THANKS to Adafruit and its GFX library !
+//https://learn.adafruit.com/adafruit-gfx-graphics-library
+
+void lcdDrawCircleQuarter(coord_t x0, coord_t y0, int16_t r, uint8_t corners, LcdFlags att)
+{
+  int16_t f = 1 - r;
+  int16_t ddF_x = 1;
+  int16_t ddF_y = -2 * r;
+  int16_t x = 0;
+  int16_t y = r;
+
+  if (corners >= 15) {
+	lcdDrawPoint(x0, y0 + r, att);
+    lcdDrawPoint(x0, y0 - r, att);
+    lcdDrawPoint(x0 + r, y0, att);
+    lcdDrawPoint(x0 - r, y0, att);
+  }
+
+  while (x < y) {
+	if (f >= 0) {
+      y--;
+      ddF_y += 2;
+      f += ddF_y;
+	}
+    x++;
+    ddF_x += 2;
+    f += ddF_x;
+    if (corners & 0x04) {
+      lcdDrawPoint(x0 + x, y0 + y, att);
+      lcdDrawPoint(x0 + y, y0 + x, att);
+    }
+    if (corners & 0x02) {
+      lcdDrawPoint(x0 + x, y0 - y, att);
+      lcdDrawPoint(x0 + y, y0 - x, att);
+    }
+    if (corners & 0x08) {
+      lcdDrawPoint(x0 - y, y0 + x, att);
+      lcdDrawPoint(x0 - x, y0 + y, att);
+    }
+    if (corners & 0x01) {
+      lcdDrawPoint(x0 - y, y0 - x, att);
+      lcdDrawPoint(x0 - x, y0 - y, att);
+    }
+  }
+}
+
+void lcdFillCircleQuarter(coord_t x0, coord_t y0, int16_t r, uint8_t corners, LcdFlags att)
+{
+  int16_t f = 1 - r;
+  int16_t ddF_x = 1;
+  int16_t ddF_y = -2 * r;
+  int16_t x = 0;
+  int16_t y = r;
+  int16_t px = x;
+  int16_t py = y;
+
+  if (corners >= 3) {
+    lcd->drawSolidVerticalLine(x0, y0 - r, 2 * r +1, att);
+  }
+  while (x < y) {
+    if (f >= 0) {
+      y--;
+      ddF_y += 2;
+      f += ddF_y;
+    }
+    x++;
+    ddF_x += 2;
+    f += ddF_x;
+    if (x < (y + 1)) {
+      if (corners & 0x01) {
+    	lcd->drawSolidVerticalLine(x0 + x, y0 - y, 2 * y + 1, att);
+      }
+      if (corners & 0x02) {
+    	lcd->drawSolidVerticalLine(x0 - x, y0 - y, 2 * y + 1, att);
+      }
+    }
+    if (y != py) {
+      if (corners & 0x01) {
+    	lcd->drawSolidVerticalLine(x0 + py, y0 - px, 2 * px + 1, att);
+      }
+      if (corners & 0x02) {
+    	lcd->drawSolidVerticalLine(x0 - py, y0 - px, 2 * px + 1, att);
+      }
+      py = y;
+    }
+    px = x;
+  }
+}
+
+void lcdFillTriangle(coord_t x0, coord_t y0, coord_t x1, coord_t y1, coord_t x2, coord_t y2, LcdFlags att)
+{
+  int32_t a, b;
+
+  #define SWAP(xx,yy) {int32_t t = xx; xx = yy; yy = t;}
```

Comment:
perhaps you could use the SWAP function from opentx (opentx_helpers.h)

Reply:
thx. Done.